Userland/Libraries/LibJS/Runtime/Completion.cpp

```diff
@@ -89,21 +90,31 @@ ThrowCompletionOr<Value> await(GlobalObject& global_object, Value value)
     auto on_rejected = NativeFunction::create(global_object, "", move(rejected_closure));
 
     // 7. Perform ! PerformPromiseThen(promise, onFulfilled, onRejected).
-    verify_cast<Promise>(promise)->perform_then(on_fulfilled, on_rejected, {});
+    auto* promise = verify_cast<Promise>(promise_object);
+    promise->perform_then(on_fulfilled, on_rejected, {});
+
+    // FIXME: Since we don't support context suspension, we attempt to "wait" for the promise to resolve
+    //        by letting the event loop spin until our promise is no longer pending, and then synchronously
+    //        running all queued promise jobs.
+    if (Core::EventLoop::has_been_instantiated())
```

Comment:
Can you also add to this comment that this is solely needed for LibWeb?

Reply:
I mean...it's useful for any embedder, but sure, I'll add that